### PR TITLE
Funding guidance

### PR DESCRIPTION
**Pull request by mattandrews**
Add an endpoint for this "listing" structure.

Down the line we may wish to make this more generic for other, similar structures.

### CODE REVIEW

File: config/element-api.php [hunk 1]
```diff
@@ -217,11 +217,148 @@ function getLegacyPage($locale)
...
+        }
+    }
+
+    return $basicData;
+}
+
+function getRelatedEntries($entry, $relationType) {
+    $relatedEntries = [];
+    if ($relationType == 'children') {
+        $relatedSearch = $entry->getChildren()->find();
```
Comment (davidrapson) [inline config/element-api.php]: `->find()` seems to be deprecated, logs mention using `->all()` instead.

File: config/element-api.php [hunk 2]
```diff
@@ -217,11 +217,148 @@ function getLegacyPage($locale)
...
+                $entryData['children'] = $children;
+            }
+            
+            $siblings = getRelatedEntries($entry, 'siblings');
+            if (count($siblings) > 0) {
+                $thisEntry['isCurrent'] = true;
+                $thisEntry['order'] = $entry->lft;
+                $entryData['siblings'] = $siblings;
+                $entryData['siblings'][] = $thisEntry;
+                // reorder the siblings now we've appended the current page
```
Comment (davidrapson) [inline config/element-api.php]: This feels a little magic, anyway we could avoid the ordering? Does getting the parent first and then using `getDescendants` work?

File: config/element-api.php [hunk 3]
```diff
@@ -217,11 +217,152 @@ function getLegacyPage($locale)
...
+
+            $children = getRelatedEntries($entry, 'children');
+            if (count($children) > 0) {
+                $entryData['children'] = $children;
+            }
+            
+            $siblings = getRelatedEntries($entry, 'siblings');
+            if (count($siblings) > 0) {
+                $thisEntry['isCurrent'] = true;
+                $thisEntry['order'] = $entry->lft;
```
Comment (davidrapson) [inline config/element-api.php]: This feels a little magic, anyway we could avoid the ordering? Does getting the parent first and then using `getDescendants` work?

Comment (mattandrews) [reply config/element-api.php]: Give this another look – have used your parent/descendants technique and it works well.

File: config/element-api.php (hunk 1 above)
Comment (mattandrews) [reply config/element-api.php]: 👍

**Pull request merged**